c2rust-analyze/src/main.rs

```diff
@@ -718,8 +809,11 @@ fn run(tcx: TyCtxt) {
     eprintln!("\nfinal labeling for static items:");
     let lcx1 = crate::labeled_ty::LabeledTyCtxt::new(tcx);
     let lcx2 = crate::labeled_ty::LabeledTyCtxt::new(tcx);
-    for (did, lty) in gacx.static_tys.iter() {
-        let name = tcx.item_name(*did);
+    let mut static_dids = gacx.static_tys.keys().cloned().collect::<Vec<_>>();
+    static_dids.sort();
```

Comment:
why sorted? add a comment?

Reply:
Determinism.  Hash tables always need to be sorted before iterating if iteration order might affect the output, since iteration order is either random or unreliable in nearly all languages and libraries.